src/components/pagination/pagination.js

```diff
@@ -92,56 +94,98 @@ class Pagination extends React.PureComponent {
         className={classnames('pagination', className)}
         aria-label="pagination"
       >
-        {
-          showPrevNext &&
-            <React.Fragment>
-              <a
-                role="button"
-                tabIndex={0}
-                onClick={current === 1 ? undefined : this.goToPage(current - 1)}
-                className="pagination-previous"
-                title="This is the first page"
-                disabled={current === 1}
-              >
-                {previous}
-              </a>
-              <a
-                role="button"
-                tabIndex={0}
-                onClick={current === total ? undefined : this.goToPage(current + 1)}
-                className="pagination-next"
-                disabled={current === total}
-              >
-                {next}
-              </a>
-            </React.Fragment>
-        }
-        {
-          delta > 0 &&
-            <React.Fragment>
-              <ul className="pagination-list">
-                {
-                  Array((lastPage - firstPage) + 1).fill(0).map((_, i) => (
-                    // eslint-disable-next-line react/no-array-index-key
-                    <li key={i + firstPage}>
+        {showPrevNext && (
+          <React.Fragment>
+            <a
+              role="button"
+              tabIndex={0}
+              onClick={current === 1 ? undefined : this.goToPage(current - 1)}
+              className="pagination-previous"
+              title="This is the first page"
+              disabled={current === 1}
+            >
+              {previous}
+            </a>
+            <a
+              role="button"
+              tabIndex={0}
+              onClick={
+                current === total ? undefined : this.goToPage(current + 1)
+              }
+              className="pagination-next"
+              disabled={current === total}
+            >
+              {next}
+            </a>
+          </React.Fragment>
+        )}
+        {delta > 0 && (
+          <React.Fragment>
+            <ul className="pagination-list">
+              {current !== 1 &&
+                current > delta + 1 && (
+                  <React.Fragment>
+                    <li>
+                      <a
+                        role="button"
+                        className="pagination-link"
+                        aria-label="Goto page 1"
+                        tabIndex={0}
+                        onClick={this.goToPage(1)}
+                      >
+                        1
```

Comment:
Do no change the default behaviour the the component, to add this feature (First/Last page) add a new Boolean prop to enable this.

Reply:
Hi I will correct that ASAP. 🤭